RF24.h

```diff
@@ -1972,6 +2056,152 @@ class RF24 {
+        }
+        // k-th bit is 0
+        return false;
+    }
+
+    /**
+     * @brief Set bit `k` of variable `A`.
+     * 
+     * @tparam T T is the template parameter; the `typename` keyword means
+     * that this parameter is a placeholder for a type.  At compile time, 
+     * this parameter `T` is expanded into every type that uses it.
+     * @param A The variable that contains the bit we want to set.
+     * @param k The position of the bit we want to set within `A`.
+     */
+    template <typename T>
+    static void SetBit(T A[], uint8_t k)
+    {
+        // where 8 is the width of one byte, else change 8 to your byte width
+        A[k / (sizeof(T) * 8)] |= _BV(k % (sizeof(T) * 8));
+    }
+
+    /**
+     * @brief Pack value `source` into the array `encoded_details` for `number_of_bits`.
+     *
+     * @tparam T T is the template parameter; the `typename` keyword means
+     * that this parameter is a placeholder for a type.  At compile time, 
+     * this parameter `T` is expanded into every type that uses it.
+     * @param source The variable we want to pack into the bit array `encoded_details`.
+     * @param number_of_bits The amount of bits we want to put into the bit array `encoded_details`.
+     * @param encoded_details The bit array.
+     * @param encoded_details_index The byte index of the bit array `encoded_details`.
+     * @param bit_index The bitwise index of the bit array `encoded_details`.
+     */
+    template <typename T>    
+    static void packValue(T source, int number_of_bits, uint8_t *encoded_details, uint8_t *encoded_details_index, uint16_t *bit_index)
+    {
+        for (uint8_t i = 0; i < number_of_bits; ++i)
+        {
+            if (TestBit(&source, i))
+            {
+                uint8_t index = (*bit_index) % 8;
+                SetBit(&encoded_details[*encoded_details_index], index);
+            }
+            (*bit_index)++;
+            if (*bit_index % 8 == 0)
+            {
+                (*encoded_details_index)++;
+            }
+        }
+    }
+
+    /**
+     * @brief Get bits from `encoded_details` and put them into `out` for `number_of_bits`.
+     *
+     * @tparam T T is the template parameter; the `typename` keyword means
+     * that this parameter is a placeholder for a type.  At compile time, 
+     * this parameter `T` is expanded into every type that uses it.
+     * @param out The variable we want to populate with bits read from the bit array `encoded_details`.
+     * @param number_of_bits The number of bits we want to read out of the bit array `encoded_details`.
+     * @param encoded_details The bit array.
+     * @param encoded_details_index The byte index of the bit array `encoded_details`.
+     * @param bit_index The bitwise index of the bit array `encoded_details`.
+     */
+    template <typename T>
+    static void getBits(T *out, int number_of_bits, uint8_t *encoded_details, uint8_t *encoded_details_index, uint16_t *bit_index)
```

Comment:
Would it be easier to make this return a value of type `T` instead of passing by reference? Most of the calls to this function allocate a new byte to pass `out` by reference, then pass `out`'s value to `sprintf()`. It could be called directly from within a `sprintf()` call... The only times that a temp byte needs to be allocated is when you perform arithmetic on the channel value (`temp_byte + 2400`) or use the value as an index to an array of translating strings.